sdk/scheduler/src/main/java/com/mesosphere/sdk/state/DefaultStateStore.java

```diff
@@ -322,22 +274,26 @@ private CuratorTransactionFinal deleteChildren(String root, CuratorTransactionFi
         Collection<Protos.TaskStatus> taskStatuses = new ArrayList<>();
         for (String taskName : fetchTaskNames()) {
             try {
-                byte[] bytes = persister.get(taskPathMapper.getTaskStatusPath(taskName));
+                byte[] bytes = persister.get(getTaskStatusPath(taskName));
                 taskStatuses.add(Protos.TaskStatus.parseFrom(bytes));
-            } catch (KeeperException.NoNodeException e) {
-                // The task node exists, but it doesn't contain a TaskStatus node. This may occur if
-                // the only contents are a TaskInfo.
-            } catch (Exception e) {
-                throw new StateStoreException(Reason.STORAGE_ERROR, e);
+            } catch (PersisterException e) {
```

Comment:
These catches were effectively moved upstream into `CuratorPersister` (including `} catch (Exception e) {`). Any exceptions thrown by Curator are translated to `PersisterException`s before they're passed upstream to `Default[State|Config]Store`.